corehq/toggles.py

```diff
@@ -220,14 +221,18 @@ def randomness_percent(self):
     def _get_identifier(self, item):
         return '{}:{}:{}'.format(self.namespaces, self.slug, item)
 
-    def enabled(self, item, **kwargs):
+    def enabled(self, item, namespace=Ellipsis):
+        if namespace is Ellipsis:
+            logging.warning('PredictablyRandomToggle.enabled() may be invalid if namespace is not set')
         if settings.UNIT_TESTING:
             return False
         elif item in self.always_disabled:
             return False
+        elif namespace is not Ellipsis and namespace not in self.namespaces:
+            return False
         return (
             (item and deterministic_random(self._get_identifier(item)) < self.randomness)
```

Comment:
I think the real problem is on this line, not the one below it. If this line evaluates to true the `super.enabled(...)` call will not be evaluated. I think we cannot evaluate the randomness check here unless the toggle applies to both users and domains or the namespace parameter is specified.

Reply:
Agreed.